nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -681,6 +686,41 @@ impl DataStore {
             .filter(service_network_interface::time_deleted.is_null())
             .select(Sled::as_select());
 
+        // ... and we also need to query for the current target blueprint to
+        // support systems that _are_ under Reconfigurator control. This query
+        // needs a join with a subquery, which isn't currently supported by
+        // diesel's DSL, so we fall back to a raw SQL string and rely on tests
+        // to ensure that this is valid.
+        //
+        // Diesel requires us to use aliases in order to refer to the
+        // `inv_collection` table twice in the same query.
+        let (bp_target1, bp_target2) = diesel::alias!(
+            db::schema::bp_target as bp_target1,
+            db::schema::bp_target as bp_target2
+        );
+        let reconfig_service_query = service_network_interface::table
+            .inner_join(bp_omicron_zone::table.on(
+                bp_omicron_zone::id.eq(service_network_interface::service_id),
+            ))
+            .inner_join(
+                bp_target1.on(bp_omicron_zone::blueprint_id
+                    .eq(bp_target1.field(bp_target::blueprint_id))),
+            )
+            .inner_join(sled::table.on(sled::id.eq(bp_omicron_zone::sled_id)))
+            .filter(
+                // This filters us down to the one current target blueprint (if
+                // it exists); i.e., the target with the maximal version.
+                bp_target1.field(bp_target::blueprint_id).eq_any(
+                    bp_target2
+                        .select(bp_target2.field(bp_target::blueprint_id))
+                        .order_by(bp_target2.field(bp_target::version).desc())
```

Comment:
I thought a bunch about whether we want to add an `enabled = true` filter here.  I think we don't.  Regardless of whether we're trying to do anything with this blueprint, it _is_ the current intended state.

Reply:
Agreed. I will add a comment to that effect.